Hi @joachimmetz  — I'm not sure why, but we have one of the compiles failing:


```
tsk/fs/ntfs.cpp: In function 'TSK_RETVAL_ENUM ntfs_make_data_run(NTFS_INFO*, TSK_OFF_T, ntfs_runlist*, uint32_t, TSK_FS_ATTR_RUN**, TSK_OFF_T*, TSK_INUM_T)':
tsk/fs/ntfs.cpp:701:85: error: comparison of integer expressions of different signedness: 'int' and 'uint32_t' {aka 'unsigned int'} [-Werror=sign-compare]
  701 |         if (NTFS_RUNL_OFFSZ(run) > 8 || NTFS_RUNL_LENSZ(run) + NTFS_RUNL_OFFSZ(run) > runlist_size - runlist_offset - 1) {
```

Any clue?

--- a/tsk/fs/ntfs.cpp
+++ b/tsk/fs/ntfs.cpp
@@ -656,7 +656,7 @@ ntfs_make_data_run(NTFS_INFO * ntfs, TSK_OFF_T start_vcn,
          * A length of more than eight bytes will not fit in the
          * 64-bit length field (and is likely corrupt)
          */
-        if (NTFS_RUNL_LENSZ(run) > 8 || NTFS_RUNL_LENSZ(run) > runlist_size - runlist_offset - 1) {
+        if (NTFS_RUNL_LENSZ(run) > 8 || (uint32_t) NTFS_RUNL_LENSZ(run) > runlist_size - runlist_offset - 1) {
             tsk_error_reset();
             tsk_error_set_errno(TSK_ERR_FS_INODE_COR);
             tsk_error_set_errstr
@@ -698,7 +698,7 @@ ntfs_make_data_run(NTFS_INFO * ntfs, TSK_OFF_T start_vcn,
          * An address offset of more than eight bytes will not fit in the
          * 64-bit addr_offset field (and is likely corrupt)
          */
-        if (NTFS_RUNL_OFFSZ(run) > 8) {
+        if (NTFS_RUNL_OFFSZ(run) > 8 || (uint32_t) ( NTFS_RUNL_LENSZ(run) + NTFS_RUNL_OFFSZ(run) ) > runlist_size - runlist_offset - 1) {
             tsk_error_reset();
             tsk_error_set_errno(TSK_ERR_FS_INODE_COR);
             tsk_error_set_errstr

